### PR TITLE
Implement code generation logic

### PR DESCRIPTION
**Pull request by OviBortas**
### What does this PR do? :soccer:

This PR adds the logic needed to create the code generation for Fonts and Colors.

### How can this PR be tested? :mag:

1. Run `npm start` on the project.
2. Open zeplin and in the menu bar go to `Extensions` > `Manage Project Extensions`. Disable all other extensions except `swift-abstracted-style` and tap `done`
3. Go to the `Styleguide` > `Colors & Text Styles` in Zeplin
4. In the right panel where code generation is shown you should see code for named and abstracted colors generated along with abstracted fonts.

### CODE REVIEW

File: dist/main.3d4d9380.js [hunk 1]
```diff
@@ -0,0 +1 @@
...K: - Abstract Colors\n    ${t.map(l).join("\n")}\n}`}(n),language:"swift",filename:"UIColor+AppColors.swift"}}function u(n){return`UIColor(red: ${n.r}/255, green: ${n.g}/255, blue: ${n.b}/255, alpha: ${n.a})`}function i(n){return`${o}static let ${n.name}: UIColor = {\n        return ${u(n)}\n    }()`}function l(n){var e=n.name,t=n.style.color;return`${o}static let ${e}: UIColor = {\n        return ${u(t)}\n    }()`}function s(n){return{code:function(n){return`\nimport UIKit\n\nextension UIFont {\n${n.map(f).join("\n")}\n}\n\nextension UILabel {\n    \n    func setFontStyle(row: UIFont, column: UIColor) {\n        font = row\n        textColor = column\n    }\n}`}(a(n,0,2).sort(function(n,e){return n.name>e.name})),language:"swift",filename:"UIFont+AppFonts.swift"}}function f(n){var e=n.name,t=n.style.fontFace,r=n.style.fontSize;return`${o}static let ${e}: UIFont = {\n        return UIFont(name: "${t}", size: ${r})!\n    }()`}function a(n,e,t){return n.map(function(n){return{name:n.name.substring(e,t).toLowerCase(),style:n}}).filter(function(n,e,t){return t.findIndex(function(e){return e.name===n.name})===e})}e.default={generateFontExtension:s,generateColorExtension:r}}]).default});
```
[the remark this answers is not in the document]
Comment (hyjeko) [reply dist/main.3d4d9380.js]: Actually since you aren't `publishing` this, it's probably okay to commit.

[the remark this answers is not in the document]
Comment (OviBortas) [reply dist/main.3d4d9380.js]: 🤦‍♂️ I used the `Swift` gitignore template. Will change.

File: src/index.js [hunk 2]
```diff
@@ -3,29 +3,27 @@
...
 
 function exportStyleguideTextStyles(context, textStyles) {
-
+    return helper.generateFontExtension(textStyles)
 }
 
-function comment(context, text) {
-
-}
+function comment(context, text) {}
```
[the remark this answers is not in the document]
Comment (OviBortas) [reply src/index.js]: Nice catch nope. Will change.

File: src/helpers.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    return {
+        code: colorExtensionTemplate(project),
+        language: 'swift',
+        filename: 'UIColor+AppColors.swift',
+    };
+}
+
+// Returns a string that represents the code for the color extension populated with the given colors.
+function colorExtensionTemplate(project) {
+    let colors = project.colors
```
[the remark this answers is not in the document]
Comment (OviBortas) [reply src/helpers.js]: Hmmm thats interesting. `var` and `let` are just scoped variables... In swift `var` means mutable while `let` mean immutable regardless of scope. Thanks for the heads up. Will change.

File: src/helpers.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    let textStyleObjects = uniqueTextStyles(project.textStyles, 2, 4)
+        .sort((a, b) => a.name > b.name)
+
+    console.log(textStyleObjects.length)
```
[the remark this answers is not in the document]
Comment (OviBortas) [reply src/helpers.js]: Removed.

File: src/index.js [hunk 5]
```diff
@@ -3,29 +3,27 @@
  * https://github.com/zeplin/zeplin-extension-documentation
  */
 
-function layer(context, selectedLayer) {
+const helper = require('./helpers')
 
-}
+function layer(context, selectedLayer) {}
```
[the remark this answers is not in the document]
Comment (OviBortas) [reply src/index.js]: Removed.

**Pull request merged**